test/gtest/uct/cuda/test_switch_cuda_device.cc

```diff
@@ -484,3 +485,157 @@ UCS_TEST_P(test_p2p_no_current_cuda_ctx, get_zcopy)
 }
 
 _UCT_INSTANTIATE_TEST_CASE(test_p2p_no_current_cuda_ctx, cuda_copy)
+
+class test_p2p_send_on_diff_device : public uct_p2p_test {
+public:
+    typedef ucs_status_t (test_p2p_send_on_diff_device::*rma_send_f_t)(
+                          uct_ep_h, const void*, size_t, uint64_t, uct_rkey_t);
+
+    using rkey_mem_pair_t = std::pair<uct_rkey_bundle_t, uct_mem_h>;
+
+    test_p2p_send_on_diff_device() : uct_p2p_test(0), m_num_devices(0) { }
+
+    template<typename mem_src_t, typename mem_dest_t = mem_buffer>
+    void send_diff_device(rma_send_f_t s_func,
+                          ucs_memory_type_t dest_type = UCS_MEMORY_TYPE_HOST)
+    {
+        int current_device;
+        ASSERT_EQ(cudaGetDevice(&current_device), cudaSuccess);
+
+        const size_t size = UCS_MBYTE;
+        mem_src_t src_buf(size, UCS_MEMORY_TYPE_CUDA);
```

Comment:
shall we pass src_mem_type as parameter?

Reply:
we do not need it for this PR. Maybe we will extend it in the future